xbmc/addons/Addon.cpp

```diff
@@ -331,7 +332,19 @@ std::string CAddon::LibPath() const
 {
   if (m_props.libname.empty())
     return "";
-  return URIUtils::AddFileToFolder(m_props.path, m_props.libname);
+
+  std::string strLibPath = URIUtils::AddFileToFolder(m_props.path, m_props.libname);
+
+  // Check if add-on library has been installed to the binaries path instead
+  std::string strSharePath = CSpecialProtocol::TranslatePath("special://xbmc/");
+  const bool bIsInSharePath = StringUtils::StartsWith(strLibPath, strSharePath);
+  if (bIsInSharePath && !CFile::Exists(strLibPath))
```

Comment:
Frankly I'm not sure I understand this whole binary addon issue. Why would the library not be located inside the addon folder?

But if just having `libname` isn't sufficient for binary addons then clearly 'libname' is the wrong solution and can just be removed entirely. Because its only purpose is *here* for resolving LibPath(). More appropriate place to handle this resolving would then be either the [factory](https://github.com/xbmc/xbmc/blob/0bea8ab957a11d1352307efead0bc8c7963275b8/xbmc/addons/AddonManager.cpp#L102) or that^ `LoadDll()`